zerver/lib/slack_data_to_zulip_data.py

```diff
@@ -279,6 +302,34 @@ def channels_to_zerver_stream(slack_data_dir: str, realm_id: int, added_users: A
     return zerver_defaultstream, zerver_stream, added_channels, zerver_subscription, \
         zerver_recipient, added_recipient
 
+def get_total_messages_and_usermessages(slack_data_dir: str, channel: str,
+                                        zerver_userprofile: List[ZerverFieldsT],
+                                        zerver_subscription: List[ZerverFieldsT],
+                                        added_recipient: AddedRecipientsT) -> Tuple[int, int]:
+    """
+    Returns:
+    1. message_id, which is total number of messages
+    2. usermessage_id, which is total number of usermessages
+    """
+    json_names = os.listdir(slack_data_dir + '/' + channel)
```

Comment:
I guess there is no way to parse through the channel directories just once. Getting the total count and converting each message to Zulip have to be done separately.

Reply:
@rht Yes! We have to repeat the process twice, once to get the total number of objects, and then we get the id range and then the second time to allocate these ids to the objects. I am concerned about the process being slowed down here.